src/NameApp.java

```diff
@@ -0,0 +1,313 @@
+        decadeChoice = keyboard.nextInt();
+
+        if (decadeChoice < 1 || decadeChoice > 11) {
+            return -1;
+        }else{
+            return decadeChoice;
+        }
+    }
+
+    //Prompts for and gets the name from the user
+    private static String getUsersNameChoice() {
+        Scanner keyboard = new Scanner(System.in);
+        String name = "";
+
+        System.out.print("Enter a name: ");
+        return keyboard.next();
+    }
+
+    private static String decadeChoice(int decade) {
+        String decadeRange = "";
+        switch (decade) {
+            case 1:
+                decadeRange = "1900-1909";
+            case 2:
+                decadeRange = "1910-1919";
+            case 3:
+                decadeRange = "1920-1929";
+            case 4:
+                decadeRange = "1930-1939";
+            case 5:
+                decadeRange = "1940-1949";
+            case 6:
+                decadeRange = "1950-1959";
+            case 7:
+                decadeRange = "1960-1969";
+            case 8:
+                decadeRange = "1970-1979";
+            case 9:
+                decadeRange = "1980-1989";
+            case 10:
+                decadeRange = "1990-1999";
+            case 11:
+                decadeRange = "2000-2005";
+        }
+        return decadeRange;
+    }
+
+    private static void writeAnomaliesToFile(Name[] nameList) {
+        Name[] temp = new Name[nameList.length];
+        String[] anomalies = new String[1065];
+        int anomalyCount = 0;
+
+        for (int i = 0; i < nameList.length; i++){
+            temp[i] = nameList[i];
+        }
+
+        for (int decade = 0; decade < 11; decade++){
+            for (int ranks = 1; ranks < 1000; ranks++){
+                int found = 0;
+                String tempName = "";
+
+                String decadeYear = decadeChoice(decade);
+
+                for (int i = 0; i < temp.length; i++){
+                    if (temp[i].getPop(decade) == ranks){
+                        found++;
+                    }
+
+                    if (found == 1 &&
+                            temp[i].getPop(decade) == ranks){
+                        tempName = temp[i].getName();
+                    }
+                }
+
+                if (found == 0){
+                    anomalies[anomalyCount] = "No names "
+                            + "for " + decadeYear + ", rank "
+                            + ranks + ".";
+                    anomalyCount++;
+                }
+                else if (found == 1){
+                    anomalies[anomalyCount] = "One "
+                            + "Name (" + tempName + ") for "
+                            + decadeYear + ", rank "
+                            + ranks + ".";
+                    anomalyCount++;
+                }
+            }
+        }
+
+        //write the anomalies to the anomalies.txt file
+        String fileName = "anomalies.txt";
```

Comment:
This is something you might consider making a constant at the beginning of the function/class as well.